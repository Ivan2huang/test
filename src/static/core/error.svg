<svg width="528" height="528" viewBox="0 0 528 528" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M376.396 62.5246C406.125 89.2346 425.581 126.502 456.816 180.067C488.197 233.632 531.47 303.489 515.059 345.137C498.649 386.784 422.264 400.208 350.142 427.519C278.019 454.831 210.117 495.87 147.653 487.87C85.1887 479.869 28.1615 422.842 13.8189 359.77C-0.669133 296.546 27.3952 227.29 47.611 164.218C67.6815 101.153 79.9035 44.2445 115.057 18.4792C150.21 -7.28607 208.307 -1.74984 257.499 8.95926C306.684 19.8005 346.667 35.8212 376.396 62.5246Z" fill="#F1F1F1"/>
<path d="M346.421 512.446H357.659V502.932H346.421V512.446Z" fill="#DE8E68"/>
<path d="M358.716 511.341V520.855H331.979V517.393L345.886 511.341H358.716Z" fill="black"/>
<path d="M171.747 395.742H182.985V386.229H171.747V395.742Z" fill="#DE8E68"/>
<path d="M184.041 394.256V403.769H157.311V400.314L171.218 394.256H184.041Z" fill="black"/>
<path d="M346.032 267.022C346.601 272.426 358.538 505.443 358.538 505.443H346.032L285.22 286.042L346.032 267.022Z" fill="black"/>
<path d="M352.453 241.349C352.453 272.994 340.819 301.468 317.267 301.468C284.935 301.468 242.132 280.855 230.352 269.221C227.413 277.916 185.712 387.425 185.712 387.425H168.443C168.443 387.425 179.509 259.92 200.253 227.217C207.078 218.127 262.77 220.4 290.907 227.217C319.044 234.035 352.453 241.349 352.453 241.349Z" fill="#666666"/>
<path d="M268.941 56.9685C269.137 55.991 269.685 55.1196 270.482 54.5203C271.279 53.9209 272.268 53.6354 273.261 53.7181C275.792 53.8965 279.927 60.5822 279.927 65.9929C279.927 71.4036 278.401 73.4384 278.864 77.3693C279.326 81.3002 289.335 134.601 289.335 137.852C291.859 138.03 300.89 136.768 300.89 136.768L294.567 162.23H276.162L264.27 59.6771L268.941 56.9685Z" fill="#D37C59"/>
<path d="M286.594 66.0461L282.716 81.7167C282.716 81.7167 244.457 77.2705 240.282 76.2399C236.107 75.2093 242.396 49.2987 246.684 50.3557C250.971 51.4127 286.594 66.0461 286.594 66.0461Z" fill="black"/>
<path d="M321.648 112.404C370.589 112.404 358.519 238.845 352.804 243.575C347.09 248.306 286.244 232.344 281.19 225.302C276.136 218.259 285.755 112.404 321.648 112.404Z" fill="#038743"/>
<path d="M304.18 144.505C301.492 145.416 285.722 151.567 285.722 151.567C285.458 148.396 273.335 85.5021 272.588 81.6505C274.118 77.8061 274.875 73.6978 274.815 69.5606C274.815 59.2347 270.62 50.8511 265.447 50.8511C260.274 50.8511 256.085 59.2347 256.085 69.5606C256.012 73.8826 256.845 78.172 258.53 82.1526C257.948 90.0143 252.729 164.661 263.379 175.304C274.61 186.535 301.914 176.407 318.51 168.863C322.064 167.634 317.222 140.977 304.18 144.505Z" fill="#DE8E68"/>
<path d="M292.469 72.5185L282.572 69.8906L280.736 76.8058L290.633 79.4337L292.469 72.5185Z" fill="black"/>
<path d="M352.837 135.632C360.295 150.113 330.758 176.202 322.586 174.26C314.413 172.318 301.233 143.157 303.698 137.924C306.162 132.692 345.583 121.54 352.837 135.632Z" fill="#025E2E"/>
<path d="M324.646 124.658C324.646 127.109 309.696 127.73 309.696 124.658V96.4355H324.646V124.658Z" fill="#D37C59"/>
<path d="M311.771 112.819C322.549 112.819 331.286 98.6895 331.286 81.2598C331.286 63.8302 322.549 49.7007 311.771 49.7007C300.993 49.7007 292.255 63.8302 292.255 81.2598C292.255 98.6895 300.993 112.819 311.771 112.819Z" fill="#DC8E6C"/>
<path d="M297.151 60.351C297.002 62.6177 297.459 64.8831 298.475 66.9148C299.491 68.9465 301.029 70.6713 302.931 71.9124C308.752 75.8763 312.901 74.1256 317.168 77.8582C319.421 80.111 324.938 84.4647 324.99 91.1901C324.99 96.1648 326.622 98.6752 328.056 98.6752C329.489 98.6752 333.268 97.6843 335.7 97.6843C359.575 97.6843 387.613 122.333 441.403 122.333C469.501 122.333 513.592 115.244 513.592 73.9802C513.592 51.6966 505.572 24.7686 463.178 24.7686C436.257 24.7686 417.395 37.9815 395.277 37.9815C366.869 37.9815 355.915 27.6952 336.288 27.6952C322.308 27.6952 290.848 35.7948 290.848 57.0148C290.848 65.0747 293.372 67.4134 294.151 67.8296C294.402 66.9443 295.942 62.3726 297.151 60.351Z" fill="black"/>
<path d="M322.982 93.7145C325.733 93.7145 327.963 91.4843 327.963 88.7332C327.963 85.9821 325.733 83.752 322.982 83.752C320.23 83.752 318 85.9821 318 88.7332C318 91.4843 320.23 93.7145 322.982 93.7145Z" fill="#DC8E6C"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M159.526 402.998C153.218 403.112 146.957 406.294 143.352 412.546L93.2889 499.346C85.9441 512.081 95.1251 527.999 109.815 527.999H209.941C224.63 527.999 233.811 512.081 226.467 499.346L176.403 412.546C175.965 411.786 175.488 411.072 174.976 410.403V402.995H159.979C159.911 402.995 159.844 402.995 159.777 402.995H159.526V402.998Z" fill="#757575"/>
<path d="M158.101 412.547C165.445 399.812 183.807 399.812 191.152 412.547L241.215 499.348C248.56 512.082 239.379 528.001 224.689 528.001H124.563C109.874 528.001 100.693 512.082 108.037 499.348L158.101 412.547Z" fill="#E9A115"/>
<path d="M181.691 481.026L168.824 481.476L165.525 444.039L182.37 443.451L181.691 481.026ZM167.476 495.416C167.392 493.011 168.028 491.123 169.386 489.752C170.743 488.355 172.76 487.61 175.437 487.516C178.088 487.424 180.114 488.028 181.512 489.329C182.936 490.602 183.69 492.442 183.774 494.847C183.857 497.226 183.207 499.114 181.825 500.512C180.467 501.883 178.515 502.612 175.967 502.701C173.394 502.791 171.356 502.201 169.854 500.93C168.352 499.633 167.559 497.795 167.476 495.416Z" fill="white"/>
</svg>
