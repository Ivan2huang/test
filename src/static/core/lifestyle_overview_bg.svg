<svg width="659" height="498" viewBox="0 0 659 498" fill="none" xmlns="http://www.w3.org/2000/svg">
  <g clip-path="url(#clip0)">
    <path d="M42.942 270.609C101.486 369.66 271.979 442.37 361.552 388.783C498.638 317.674 700.852 68.0021 639.174 4.805C577.496 -58.3922 389.515 -295.208 176.898 -169.283C-35.7203 -43.3576 -15.6017 171.557 42.942 270.609Z" fill="#EAEAEA"/>
    <path d="M246.354 288.285C200.735 287.817 151.644 297.828 132.801 302.892C91.8538 316.948 164.275 324.276 222.18 324.132C280.085 323.988 363.661 319.817 388.597 303.256C402.374 294.107 303.378 288.871 246.354 288.285Z" fill="#D1D1D1"/>
    <g clip-path="url(#clip1)">
      <path d="M188.413 148.564H320.308" stroke="#DEDEDE" stroke-width="9" stroke-miterlimit="10"/>
      <path d="M166.004 148.564H171.526" stroke="#DEDEDE" stroke-width="15" stroke-miterlimit="10"/>
      <path d="M337.482 148.564H343.004" stroke="#DEDEDE" stroke-width="15" stroke-miterlimit="10"/>
      <path d="M244.767 210.841C242.026 199.908 233.805 187.804 230.673 184.095C227.541 180.385 219.124 161.253 211.686 151.297C209.533 151.883 207.967 152.663 207.967 153.64C207.967 154.616 210.316 170.039 219.516 186.437C221.32 189.653 227.15 215.136 238.111 221.578C243.126 224.52 244.767 210.841 244.767 210.841Z" fill="#DE8E68"/>
      <path d="M263.95 210.841C266.691 199.908 274.912 187.804 278.044 184.095C281.176 180.385 289.593 161.253 297.031 151.297C299.184 151.883 300.75 152.663 300.75 153.64C300.75 154.616 298.401 170.039 289.201 186.437C287.398 189.653 281.567 215.136 270.606 221.578C265.6 224.52 263.95 210.841 263.95 210.841Z" fill="#DE8E68"/>
      <path d="M299.38 153.835C302.299 153.835 304.665 151.475 304.665 148.564C304.665 145.653 302.299 143.293 299.38 143.293C296.461 143.293 294.095 145.653 294.095 148.564C294.095 151.475 296.461 153.835 299.38 153.835Z" fill="#DE8E68"/>
      <path d="M207.968 153.835C210.887 153.835 213.253 151.475 213.253 148.564C213.253 145.653 210.887 143.293 207.968 143.293C205.049 143.293 202.683 145.653 202.683 148.564C202.683 151.475 205.049 153.835 207.968 153.835Z" fill="#DE8E68"/>
      <path d="M254.015 202.25C232.092 202.25 228.471 261.11 233.267 265.893C238.062 270.676 273.296 269.31 276.722 265.893C280.147 262.477 277.309 202.25 254.015 202.25Z" fill="#DE8E68"/>
      <path d="M245.452 188.39C246.912 188.39 248.095 187.21 248.095 185.755C248.095 184.299 246.912 183.119 245.452 183.119C243.993 183.119 242.81 184.299 242.81 185.755C242.81 187.21 243.993 188.39 245.452 188.39Z" fill="#DE8E68"/>
      <path d="M263.266 188.39C264.725 188.39 265.908 187.21 265.908 185.755C265.908 184.299 264.725 183.119 263.266 183.119C261.806 183.119 260.623 184.299 260.623 185.755C260.623 187.21 261.806 188.39 263.266 188.39Z" fill="#DE8E68"/>
      <path d="M274.194 227.651C269.854 226.159 266.262 223.047 264.175 218.97C262.088 214.892 261.668 210.166 263.003 205.786C260.419 203.556 257.438 202.254 254.014 202.254C250.835 202.254 248.042 203.496 245.597 205.627C246.334 207.767 246.637 210.031 246.488 212.288C246.338 214.545 245.74 216.75 244.727 218.775C243.715 220.799 242.308 222.602 240.59 224.078C238.871 225.554 236.875 226.675 234.718 227.374C230.531 244.011 230.607 263.248 233.259 265.897C238.055 270.68 273.289 269.314 276.715 265.897C278.625 264.004 278.594 244.553 274.194 227.651Z" fill="black"/>
      <path d="M248.263 257.953C247.566 256.264 232.915 235.631 210.289 235.631C207.85 235.631 222.597 272.528 222.597 275.832C222.597 279.135 250.655 284.031 254.365 284.031V257.732L248.263 257.953Z" fill="#025E2E"/>
      <path d="M209.238 235.621C206.949 235.621 205.115 238.14 205.115 241.382C205.115 258.865 228.24 296.436 229.942 296.436H239.257C233.263 276.476 230.732 235.621 209.238 235.621Z" fill="#038743"/>
      <path d="M260.455 257.953C261.154 256.264 275.805 235.631 298.429 235.631C300.868 235.631 286.121 272.528 286.121 275.832C286.121 279.135 258.061 284.031 254.352 284.031V257.732L260.455 257.953Z" fill="#025E2E"/>
      <path d="M299.123 235.621C301.411 235.621 303.245 238.14 303.245 241.382C303.245 258.865 280.12 296.436 278.419 296.436H269.104C275.097 276.476 277.626 235.621 299.123 235.621Z" fill="#038743"/>
      <path d="M238.945 305L239.258 296.43H229.942L217.62 302.523V305H238.945Z" fill="black"/>
      <path d="M269.417 305L269.104 296.43H278.419L290.739 302.523V305H269.417Z" fill="black"/>
      <path d="M258.861 208.498C258.861 211.036 249.856 211.036 249.856 208.498V194.832H258.861V208.498Z" fill="#D37C59"/>
      <path d="M264.427 184.79C265.11 184.839 267.277 180.76 267.277 175.505C267.277 174.138 266.494 167.306 263.167 167.306C262.188 167.306 261.601 170.234 261.797 171.015C261.797 169.453 258.001 160.473 252.597 160.473C250.443 160.473 250.052 163.011 250.052 163.791C250.052 166.134 250.248 168.477 251.031 169.843C250.248 168.672 248.668 165.744 245.745 165.744C244.18 165.744 242.809 166.409 242.809 167.306C242.809 169.691 244.767 171.796 245.354 172.381C244.767 171.991 244.011 171.6 242.222 171.6C240.997 171.6 240.265 173.553 240.265 174.529C240.265 178.884 242.251 183.853 244.473 184.44C246.695 185.028 257.453 177.282 257.453 177.282C257.453 177.282 261.965 184.614 264.427 184.79Z" fill="#000000"/>
      <path d="M254.358 197.273C259.71 197.273 264.048 191.985 264.048 185.462C264.048 178.938 259.71 173.65 254.358 173.65C249.007 173.65 244.669 178.938 244.669 185.462C244.669 191.985 249.007 197.273 254.358 197.273Z" fill="#DE8E68"/>
      <path d="M264.048 183.119C262.911 182.923 250.424 179.28 248.557 174.741C246.69 170.202 254.625 172.375 254.625 172.375L262.872 175.868L264.048 183.119Z" fill="#000000"/>
      <path d="M188.411 117H178.7V180.13H188.411V117Z" fill="#DEDEDE"/>
      <path d="M330.018 117H320.308V180.13H330.018V117Z" fill="#DEDEDE"/>
      <path d="M178.095 130.287H171.369V166.845H178.095V130.287Z" fill="#DEDEDE"/>
      <path d="M337.504 130.287H330.778V166.845H337.504V130.287Z" fill="#DEDEDE"/>
    </g>
  </g>
  <defs>
    <clipPath id="clip0">
      <rect width="646" height="341" fill="white" transform="translate(0 80)"/>
    </clipPath>
    <clipPath id="clip1">
      <rect width="177" height="188" fill="white" transform="translate(166.004 117)"/>
    </clipPath>
  </defs>
</svg>
