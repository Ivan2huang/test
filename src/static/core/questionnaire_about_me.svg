<svg width="215" height="215" viewBox="0 0 215 215" fill="none"
  xmlns="http://www.w3.org/2000/svg">
  <path d="M158.466 153.361C144.779 159.832 127.92 157.683 112.575 155.214C97.2295 152.745 83.4185 150 67.7992 142.722C52.213 135.442 34.8437 123.655 29.274 107.416C23.6714 91.1695 29.8722 70.4455 41.586 54.591C53.2669 38.7291 70.4539 27.768 88.9828 22.3705C107.537 17.0133 127.394 17.2092 138.152 27.5761C148.91 37.9431 150.543 58.4371 158.177 74.9621C165.806 91.5187 179.427 104.102 180.99 117.976C182.592 131.821 172.119 146.884 158.466 153.361Z" fill="#F2F2F2"/>
  <path d="M81.9799 170.168H126.967L139.039 199.001H72.3398L81.9799 170.168Z" fill="#D7D8D6"/>
  <path d="M119.156 113.137C117.558 113.167 116.152 111.289 116.015 108.944C115.878 106.598 117.063 104.673 118.661 104.643C120.258 104.614 121.665 106.491 121.801 108.837C121.938 111.182 120.754 113.108 119.156 113.137Z" fill="#DE8E68"/>
  <path d="M118.589 72.7681C119.026 73.6542 122.143 83.5757 122.102 92.1027C122.06 100.63 120.942 105.456 120.981 106.126C121.02 106.796 116.302 106.612 116.279 106.213C116.255 105.814 116.762 93.6704 116.376 91.522C115.989 89.3737 111.743 76.9822 111.746 75.4494C111.749 73.9165 118.589 72.7681 118.589 72.7681Z" fill="#DE8E68"/>
  <path d="M91.062 175.102H86.6328V178.141H91.062V175.102Z" fill="#DE8E68"/>
  <path d="M119.982 175.102H114.806V178.141H119.982V175.102Z" fill="#DE8E68"/>
  <path d="M114.615 106.335C114.736 108.628 120.138 175.101 120.138 175.101H114.615L100.25 125.181C100.25 125.181 103.984 105.206 104.245 105.258C105.478 105.571 112.235 106.769 114.615 106.335Z" fill="black"/>
  <path d="M106.555 105.759L91.6698 175.637H86.3373C86.3373 175.637 85.8684 99.0547 86.3373 99.0547C91.3397 101.174 100.789 104.648 106.555 105.759Z" fill="#212121"/>
  <path d="M111.962 60.9308C106.994 61.9721 107.445 74.4127 108.863 75.7948C110.656 77.565 117.84 76.4807 120.525 74.3475C121.767 73.3313 116.042 60.0844 111.962 60.9308Z" fill="#025E2E"/>
  <path d="M99.3818 55.1992C78.7121 55.1992 83.6129 101.958 84.933 104.685C86.253 107.412 112.513 108.471 114.597 106.335C116.682 104.199 120.52 55.1992 99.3818 55.1992Z" fill="#038743"/>
  <path d="M102.942 49.832H96.6895V58.9163C96.6895 60.08 102.942 60.08 102.942 58.9163V49.832Z" fill="#D37C59"/>
  <path d="M99.7465 53.0957C103.68 53.0957 106.868 47.7454 106.868 41.1455C106.868 34.5456 103.68 29.1953 99.7465 29.1953C95.8134 29.1953 92.625 34.5456 92.625 41.1455C92.625 47.7454 95.8134 53.0957 99.7465 53.0957Z" fill="#DE8E68"/>
  <path d="M98.5653 26.5211C94.9698 26.5211 90.2105 28.258 86.754 28.258C83.6275 28.258 77.9477 26 74.9428 26C73.0321 26 64 26.8858 64 34.1637C64 37.4465 67.4739 40.5904 68.1687 41.4588C68.8635 42.3273 70.1314 47.1908 73.3795 47.1908C77.913 47.1908 81.0395 43.8906 85.1908 43.8906C88.0567 43.8906 88.8036 44.5853 90.2279 44.5853C91.2701 44.5853 92.8333 44.099 92.8333 44.0643C92.8333 43.5258 92.7291 39.0271 93.3544 38.2976C94.2229 37.2554 105.287 38.8013 105.287 35.1363C106.069 35.8311 106.798 38.7839 106.798 39.3919C107.302 38.8708 109.178 36.6128 109.178 34.181C109.161 31.9577 107.597 26.5211 98.5653 26.5211Z" fill="black"/>
  <path d="M78.3699 112.66C80.1818 112.932 81.9616 111.085 82.3452 108.533C82.7289 105.981 81.5711 103.691 79.7592 103.419C77.9473 103.147 76.1675 104.994 75.7839 107.546C75.4002 110.098 76.558 112.388 78.3699 112.66Z" fill="#DE8E68"/>
  <path d="M82.8121 68.9792C82.2381 69.8765 77.79 80.1857 77.0121 89.4483C76.2343 98.7109 77.0201 104.116 76.9116 104.837C76.8031 105.559 82.103 106.057 82.1676 105.628C82.2321 105.198 82.8378 91.9401 83.4784 89.6652C84.1191 87.3903 90.0693 74.5669 90.2142 72.9024C90.359 71.238 82.8121 68.9792 82.8121 68.9792Z" fill="#DE8E68"/>
  <path d="M92.2442 58.6968C87.7797 56.4975 80.6229 66.3643 80.8765 68.2994C81.1951 70.7361 87.3572 74.2621 90.7153 74.2401C92.2999 74.2149 95.9058 60.512 92.2442 58.6968Z" fill="#025E2E"/>
  <path d="M86.4063 177.445V178.783L81.126 188.683H84.5651L91.4434 182.448V177.445H86.4063Z" fill="black"/>
  <path d="M114.632 177.234V182.254H129.309V180.274L119.999 177.234H114.632Z" fill="black"/>
  <path d="M112.161 190.836H97.9177C97.6282 192.747 98.6125 196.689 104.865 197.176C111.119 197.662 112.334 193.152 112.161 190.836Z" fill="white"/>
  <rect x="136" y="37" width="4" height="133" fill="#D7D8D6"/>
</svg>
