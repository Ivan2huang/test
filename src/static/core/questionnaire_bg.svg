<svg width="659" height="442" viewBox="0 0 659 442" fill="none" xmlns="http://www.w3.org/2000/svg">
  <g clip-path="url(#clip0)">
    <path d="M42.942 270.609C101.486 369.66 271.979 442.37 361.552 388.783C498.638 317.674 700.852 68.0021 639.174 4.805C577.496 -58.3922 389.515 -295.208 176.898 -169.283C-35.7203 -43.3576 -15.6017 171.557 42.942 270.609Z" fill="#EAEAEA"/>
    <path d="M370.69 162.903C370.69 152.828 364.509 144.656 356.892 144.656C349.274 144.656 343.118 152.828 343.118 162.903C343.118 168.198 344.821 172.979 347.541 176.294C346.589 183.347 344.136 193.221 343.718 194.989C343.718 194.989 366.101 210.607 366.101 196.385C366.101 185.594 366.851 178.752 367.532 174.561C369.65 171.045 370.744 167.008 370.69 162.903Z" fill="#DC8E6C"/>
    <path d="M361.496 145.896H342.549C341.494 145.896 340.64 146.751 340.64 147.806V173.746C340.64 174.8 341.494 175.655 342.549 175.655H361.496C362.551 175.655 363.406 174.8 363.406 173.746V147.806C363.406 146.751 362.551 145.896 361.496 145.896Z" fill="#EFEFEF"/>
    <path d="M263.062 229.968C267.822 228.265 302.528 228.265 330.765 245.958C332.468 239.152 338.931 190.295 338.931 190.295L367.854 192.184C367.854 192.184 374.317 255.469 361.39 288.13C355.264 301.062 284.815 274.18 284.815 274.18L263.062 229.968Z" fill="#9E9E9E"/>
    <path d="M233.892 219.447C321.293 216.606 317.202 419.777 307.771 428.392C298.341 437.006 180.984 438.397 173.191 429.399C165.397 420.402 147.352 222.264 233.892 219.447Z" fill="#666666"/>
    <path d="M247.62 220.698C238.713 218.315 227.886 219.58 221.771 221.202C225.539 232.467 243.433 369.999 245.141 435.426C251.398 435.32 260.607 435.098 268.139 434.645C267.66 384.665 264.375 274.442 247.62 220.698Z" fill="white"/>
    <path d="M247.826 229.65C247.826 233.681 224.44 233.681 224.44 229.278C224.44 224.875 221.297 192.209 221.297 192.209L244.582 190.295C244.582 190.295 247.821 227.091 247.826 229.65Z" fill="#D37C59"/>
    <path d="M276.391 198.664C276.587 196.407 276.693 194.13 276.693 191.782C276.693 160.694 258.31 135.484 235.635 135.484C212.959 135.484 194.576 160.674 194.576 191.782C194.577 195.269 194.812 198.751 195.281 202.205L276.391 198.664Z" fill="black"/>
    <path d="M268.558 167.384C263.52 165.399 258.638 163.263 254.668 159.293C244.28 170.719 226.059 184.129 219.429 188.452C222.079 205.581 232.22 218.357 244.331 218.357C258.462 218.357 269.918 200.991 269.918 179.565C269.923 175.479 269.49 171.403 268.628 167.409L268.558 167.384Z" fill="#DC8E6C"/>
    <path d="M222.364 190.38C225.647 190.38 228.308 187.718 228.308 184.435C228.308 181.152 225.647 178.49 222.364 178.49C219.08 178.49 216.419 181.152 216.419 184.435C216.419 187.718 219.08 190.38 222.364 190.38Z" fill="#DC8E6C"/>
    <path d="M268.437 310.672L217.418 291.668L189.426 366.816L240.445 385.82L268.437 310.672Z" fill="white"/>
    <path d="M256.556 302.237L231.932 293.064L229.172 300.472L253.797 309.644L256.556 302.237Z" fill="#038743"/>
    <path d="M261.905 312.657L221.068 297.445L196.312 363.907L237.148 379.118L261.905 312.657Z" fill="#025E2E"/>
    <path d="M237.518 339.287C244.355 338.929 251.685 332.436 258.571 332.436C266.813 332.436 272.602 337.655 272.602 344.184C272.602 357.962 261.836 361.967 257.755 361.967C250.904 361.967 249.75 360.914 248.476 360.013C246.4 360.516 243.347 363.111 235.211 363.111C227.075 363.111 237.518 339.287 237.518 339.287Z" fill="#DC8E6C"/>
    <path d="M189.507 236.689C153.92 228.321 140.973 352.055 146.646 362.756C152.318 373.456 232.898 369.914 237.412 369.058C238.712 364.02 240.249 340.378 238.672 337.541C236.455 336.67 197.593 336.191 197.593 336.191C198.601 332.161 211.231 281.067 211.659 275.233C213.644 247.853 209.462 241.384 189.507 236.689Z" fill="#9E9E9E"/>
    <path d="M362.391 140.047H341.021C338.914 140.047 337.207 141.754 337.207 143.861V143.866C337.207 145.972 338.914 147.679 341.021 147.679H362.391C364.497 147.679 366.205 145.972 366.205 143.866V143.861C366.205 141.754 364.497 140.047 362.391 140.047Z" fill="#038743"/>
    <path d="M360.205 168.636L353.923 172.264C353.265 172.643 353.039 173.484 353.419 174.142L353.422 174.147C353.801 174.804 354.643 175.03 355.3 174.65L361.583 171.023C362.241 170.643 362.466 169.802 362.086 169.144L362.084 169.14C361.704 168.482 360.863 168.257 360.205 168.636Z" fill="#212121"/>
    <path d="M361.806 164.006L355.248 160.904C354.562 160.58 353.742 160.873 353.417 161.56L353.415 161.564C353.09 162.251 353.383 163.071 354.07 163.396L360.628 166.497C361.315 166.822 362.135 166.528 362.46 165.842L362.462 165.837C362.786 165.15 362.493 164.33 361.806 164.006Z" fill="#212121"/>
    <path d="M341.471 166.772L343.547 173.723C343.764 174.451 344.53 174.865 345.258 174.647L345.263 174.646C345.991 174.429 346.405 173.662 346.187 172.935L344.111 165.983C343.894 165.256 343.128 164.842 342.4 165.059L342.395 165.061C341.667 165.278 341.253 166.044 341.471 166.772Z" fill="#212121"/>
    <path d="M345.219 167.942L349.863 173.516C350.349 174.099 351.216 174.178 351.8 173.692L351.804 173.689C352.387 173.202 352.466 172.335 351.98 171.752L347.336 166.178C346.85 165.595 345.983 165.516 345.399 166.002L345.395 166.005C344.812 166.491 344.733 167.359 345.219 167.942Z" fill="#212121"/>
    <path d="M350.742 168.623L357.979 169.114C358.737 169.165 359.393 168.593 359.445 167.835L359.445 167.83C359.496 167.072 358.924 166.416 358.166 166.365L350.928 165.874C350.17 165.822 349.514 166.395 349.463 167.153L349.462 167.158C349.411 167.916 349.984 168.572 350.742 168.623Z" fill="#212121"/>
  </g>
  <defs>
    <clipPath id="clip0">
      <rect width="645" height="356" fill="white" transform="translate(0 80)"/>
    </clipPath>
  </defs>
</svg>
