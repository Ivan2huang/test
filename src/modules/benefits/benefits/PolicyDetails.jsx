import React from 'react';
import PropTypes from 'prop-types';

import { Card, Box, withStyles } from '@material-ui/core';

import { injectIntl } from 'react-intl';
import GridItem from '../../../uiComponents/GridItem';
import Typography from '../../../uiComponents/Typography';
import Grid from '../../../uiComponents/Grid';
import { formatDate, formatMessage } from '../../../helpers/helpers';

const Styles = theme => ({
  [theme.breakpoints.down('sm')]: {
    itemBorder: {
      borderBottom: `1px solid ${theme.grey1}`,
    },
  },
});

const PolicyDetails = ({
  insurer,
  policyNumber,
  effectiveFrom,
  effectiveTo,
  intl,
  classes,
}) => {
  const renderDetail = (key, value, { itemBorder }) => (
    <>
      <GridItem columns={{ xs: 12, sm: 12, md: 3 }}>
        <Box pt={{ xs: 6, md: 2 }}>
          <Typography type="style-6">
            {formatMessage(intl, `me.tabs.benefits.policyDetails.${key}`, key)}
          </Typography>
        </Box>
      </GridItem>
      <GridItem columns={{ xs: 12, sm: 12, md: 9 }}>
        <Box pt={{ xs: 1, md: 2 }} pb={{ xs: 6, md: 0 }} className={itemBorder}>
          <Typography type="style-5">{value}</Typography>
        </Box>
      </GridItem>
    </>
  );

  return (
    <Card>
      <Box pl={{ md: 12 }} pr={{ md: 12 }} pt={{ md: 10 }} pb={{ md: 13 }}>
        <Grid>
          <GridItem columns={{ xs: 12, sm: 12, md: 12 }}>
            <Box mb={{ md: 6 }}>
              <Typography type="style-3">
                {formatMessage(
                  intl,
                  'me.tabs.benefits.label.policyDetails',
                  'Policy Details',
                )}
              </Typography>
            </Box>
          </GridItem>
          {renderDetail('insurer', insurer, classes)}
          {renderDetail('policyNumber', policyNumber, classes)}
          {renderDetail(
            'effectivePeriod',
            `${formatDate(effectiveFrom)} - ${formatDate(effectiveTo)}`,
            classes,
          )}
        </Grid>
      </Box>
    </Card>
  );
};

PolicyDetails.propTypes = {
  insurer: PropTypes.string.isRequired,
  policyNumber: PropTypes.string.isRequired,
  effectiveFrom: PropTypes.string.isRequired,
  effectiveTo: PropTypes.string.isRequired,
  intl: PropTypes.shape({}).isRequired,
  classes: PropTypes.exact({ itemBorder: PropTypes.string.isRequired })
    .isRequired,
};

export default injectIntl(withStyles(Styles)(PolicyDetails));
