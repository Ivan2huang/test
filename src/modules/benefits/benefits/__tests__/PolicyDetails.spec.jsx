/* eslint-disable eslint-disable-next-line,react/no-array-index-key,react/prop-types,jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */
import React from 'react';
import { withTheme } from '@material-ui/core';
import { render } from '@testing-library/react';

import PolicyDetails from '../PolicyDetails';
import withIntl from '../../../../i18n/withIntlProvider';

jest.mock(
  '../../../../uiComponents/Typography',
  () => ({ children, ...rest }) => (
    <div {...rest}>
      {children}
      (Typography)
    </div>
  ),
);

jest.mock('../../../../uiComponents/Grid', () => ({ children, ...rest }) => (
  <div {...rest}>
    Grid
    {children}
  </div>
));
jest.mock(
  '../../../../uiComponents/GridItem',
  () => ({ children, ...rest }) => (
    <div {...rest}>
      GridItem
      {children}
    </div>
  ),
);
jest.mock('../../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
  formatDate: jest.fn(date => date),
}));

describe('Policy Details', () => {
  const props = {
    insurer: 'Axa',
    policyNumber: '1234',
    effectiveFrom: '20 Aug 2018',
    effectiveTo: '29 Aug 2019',
  };
  it('should render Policy Details Component', () => {
    const Component = withIntl(withTheme(PolicyDetails));
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });
});
