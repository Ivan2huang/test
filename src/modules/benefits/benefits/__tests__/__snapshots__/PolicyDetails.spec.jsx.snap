// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Policy Details should render Policy Details Component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-31"
    >
      <div>
        Grid
        <div
          columns="[object Object]"
        >
          GridItem
          <div
            class="MuiBox-root MuiBox-root-32"
          >
            <div
              type="style-3"
            >
              Policy Details
              (Typography)
            </div>
          </div>
        </div>
        <div
          columns="[object Object]"
        >
          GridItem
          <div
            class="MuiBox-root MuiBox-root-33"
          >
            <div
              type="style-6"
            >
              insurer
              (Typography)
            </div>
          </div>
        </div>
        <div
          columns="[object Object]"
        >
          GridItem
          <div
            class="MuiBox-root MuiBox-root-34 PolicyDetails-itemBorder-1"
          >
            <div
              type="style-5"
            >
              Axa
              (Typography)
            </div>
          </div>
        </div>
        <div
          columns="[object Object]"
        >
          GridItem
          <div
            class="MuiBox-root MuiBox-root-35"
          >
            <div
              type="style-6"
            >
              policyNumber
              (Typography)
            </div>
          </div>
        </div>
        <div
          columns="[object Object]"
        >
          GridItem
          <div
            class="MuiBox-root MuiBox-root-36 PolicyDetails-itemBorder-1"
          >
            <div
              type="style-5"
            >
              1234
              (Typography)
            </div>
          </div>
        </div>
        <div
          columns="[object Object]"
        >
          GridItem
          <div
            class="MuiBox-root MuiBox-root-37"
          >
            <div
              type="style-6"
            >
              effectivePeriod
              (Typography)
            </div>
          </div>
        </div>
        <div
          columns="[object Object]"
        >
          GridItem
          <div
            class="MuiBox-root MuiBox-root-38 PolicyDetails-itemBorder-1"
          >
            <div
              type="style-5"
            >
              20 Aug 2018 - 29 Aug 2019
              (Typography)
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
